model/facade/action/PartialUpdateTasksAction.php

```diff
@@ -127,6 +127,22 @@ protected function doExecute() {
                 $discardedTasks[] = $task;
                 unset($this->tasks[$i]);
             }
+
+            if ($task->isInitDirty()){
+                // Check if init was updated and endTime should be converted to 24:00
+                // (case it was a 0-hour task and it's not anymore)
+                $currentEndTime = $oldTask->getEnd();
+                $newInitTime = $task->getInit();
+                if ($newInitTime > $currentEndTime && $currentEndTime === 0) {
+                    $task->setEnd(1440);
+                }
+
+                // Check if init was updated and endTime should be converted to 0h
+                // (case it was a regular task and now it's a 0-hour task)
+                if ($newInitTime === 0 && $currentEndTime == 1440) {
+                    $task->setEnd(0);
+                }
```

Comment:
Sorry, a side effect of my last request is that this block doesn't run when you update *the end time only*. So, if you create a task from 0:00 to 1:00, save, then update end time to 0:00, it will still be recorded as a 24:00 task.

Reply:
I think I've covered it now, hopefully I covered all scenarios 😅